### PR TITLE
Add AllEvents persistent query command

### LINKED ISSUE #4519: Akka.Persistence.Query: add "all events" query
In order to make it feasible to do things like continuously query all events, or query events that have never been tagged previously, we should add an "All Events" query that steps through the entire journal sequentially.

This query should have pagingation and live query semantics that are identical to the `EventsByTag` query. 

This query will not guarantee global order - it's going to read whatever was stored inside the journal as-is in that order.

### PR DESCRIPTION
**Pull request by Arkatufus**
Closes #4519

### CODE REVIEW

File: src/contrib/persistence/Akka.Persistence.Sql.Common/Journal/QueryExecutor.cs [hunk 1]
```diff
@@ -556,6 +585,46 @@ public virtual async Task<ImmutableArray<string>> SelectAllPersistenceIdsAsync(D
...
+                AddParameter(command, "@Ordering", DbType.Int64, fromOffset);
+                AddParameter(command, "@Take", DbType.Int64, max);
+
+                var commandBehavior = Configuration.UseSequentialAccess ? 
+                    CommandBehavior.SequentialAccess : 
+                    CommandBehavior.Default;
+
+                using (var reader = await command.ExecuteReaderAsync(commandBehavior, cancellationToken))
+                {
+                    var rowCounter = 0;
```
Comment (Aaronontheweb) [inline src/contrib/persistence/Akka.Persistence.Sql.Common/Journal/QueryExecutor.cs:607]: Might want to make this a `long`

File: src/contrib/persistence/Akka.Persistence.Query.Sql/AllEventsPublisher.cs [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        {
+            var limit = _maxBufferSize - _buffer.Length;
+            _log.Debug("replay all events request from [{0}], limit [{1}]", _currentOffset, limit);
+            _journalRef.Tell(new ReplayAllEvents(_currentOffset, limit, Self));
+            Context.Become(Replaying());
+        }
+
+        private Receive Replaying()
+        {
+            return message => message.Match()
```
Comment (Aaronontheweb) [inline src/contrib/persistence/Akka.Persistence.Query.Sql/AllEventsPublisher.cs:75]: Probably should get rid of our `PatternMatch` usage here and just go with a `switch` statement

Comment (Aaronontheweb) [reply src/contrib/persistence/Akka.Persistence.Query.Sql/AllEventsPublisher.cs:75]: With the `default` returning `false` so the system knows it was unhandled

File: src/core/Akka.Persistence.TCK/Query/AllEventsSpec.cs [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+            var eventSrc2 = queries.AllEvents(offs);
+            var probe2 = eventSrc2.RunWith(this.SinkProbe<EventEnvelope>(), Materializer);
+            probe2.Request(10);
+            // hello and world is not included, since exclusive offset
+            probe2.ExpectNext<EventEnvelope>(p => p.PersistenceId == "b" && p.SequenceNr == 1L && p.Event.Equals("test"));
+            probe2.Cancel();
+        }
+
+        [Fact]
+        public virtual void ReadJournal_query_AllEvents_should_see_all_150_events()
```
Comment (Aaronontheweb) [inline src/core/Akka.Persistence.TCK/Query/AllEventsSpec.cs:108]: Is 150 below the pagination threshold?

**Comment by Arkatufus** [comment]
Forgot to make the appropriate changes to BatchingSqlJournal

File: src/core/Akka.Persistence.TCK/Query/AllEventsSpec.cs (hunk 3 above)
Comment (Arkatufus) [reply src/core/Akka.Persistence.TCK/Query/AllEventsSpec.cs:108]: No, the command uses the max buffer size as the pagination size, it is set to 3 in the test.

Comment (Aaronontheweb) [reply src/core/Akka.Persistence.TCK/Query/AllEventsSpec.cs:108]: Ok good, so just so I understand correctly - this spec forces the journal to work through several pages worth of results in separate queries, right?

**Comment by Aaronontheweb** [comment]
cc @ismaelhamed - see anything in this query that scares you? Wanted to add support for querying all existing messages in the journal with proper pagination and offset support, just like `EventsByTag` query. Everything looks good to me so far.

**Comment by ismaelhamed** [comment]
LGTM too

File: src/core/Akka.Persistence.TCK/Query/AllEventsSpec.cs (hunk 3 above)
Comment (Arkatufus) [reply src/core/Akka.Persistence.TCK/Query/AllEventsSpec.cs:108]: yes, it retrieves the events through separate queries, using "WHERE ordering > lastindex" and "LIMIT x"

**Comment by Arkatufus** [comment]
#4518 in documentation, right? will do.

**Comment by Aaronontheweb** [comment]
@Arkatufus yep, that's correct

**Comment by Arkatufus** [comment]
PS: These changes still need to be propagated to the rest of Akka.Persistence.Sql plugin family

**Comment by Aaronontheweb** [comment]
whoops, merged this in before you added docs - could you do that in a separate PR?

**Comment by Arkatufus** [comment]
Documentation and code improvement is in #4523

**Pull request merged**